### PR TITLE
[CHI-397] Simplified Popover components

### PR DESCRIPTION
**Pull request by janibolkvadze**
@mattnickles @jllr

### Commit summary

- Introduced m-popover__content class instead of padding utility classes
- Replaced h5 with m-popover__title
- Introduced p.m-popover__text
- Introduced m-popover__arrow class for popover arrow 

#### Custom elements

- Hid m-popover__content, m-popover__text, m-popover__title classes from markup
- Added title="" attribute feature
- Added variant="" attribute feature to differentiate popover content { text, custom }

#### Tests

- Added a test case for Popover with non-text content

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/395/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/396/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/397/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/398/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/399/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/400/ for more details.

File: src/website/views/custom-elements/popover.pug [hunk 1]
```diff
@@ -168,178 +154,118 @@ ul#a-tabsb--positioning.a-tabs
...
       button#example-4-te-button.a-btn.-mr--2.-mb--2.-mb-md--0 Top end popover
-      chi-popover#example-4-te-popover(position="top-end", arrow, reference="#example-4-te-button")
-        .-px--2.-py--1
-          h5 Popover!
-          p This is popover is top-end positioned.
+      chi-popover#example-4-te-popover(position="top-end", title="Popover!", variant="text", arrow, reference="#example-4-te-button")
+        | This is popover is top-end positioned.
 
     :code(lang="html")
       <button class="a-btn -mr--2 -mb--2 -mb-md--0" id="example-4-t-button">Top popover</button>
```
Comment (mattnickles) [inline src/website/views/custom-elements/popover.pug]: Can we remove all these spacing utility classes from the Positioning and Interaction `code` examples? We're using them to provide sufficient padding in the documentation example but they're not relevant to the component.

**Comment by mattnickles** [comment]
@janibolkvadze One minor thing - can we display Methods in a table similar to Properties and Events?

File: src/website/views/custom-elements/popover.pug (hunk 1 above)
Comment (janibolkvadze) [reply src/website/views/custom-elements/popover.pug]: Makes sense, done!

Comment (janibolkvadze) [reply src/website/views/custom-elements/popover.pug]: I also changed the way CE methods are displayed for all the custom elements (Now they are shown using a table, similar to Properties and Events).

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/414/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/425/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/426/ for more details.

File: src/website/views/javascript/popover.pug [hunk 2]
```diff
@@ -7,9 +7,9 @@ p.-text
   | you can initialize it with the <code>chi.popover</code> factory method.
 .example.-mb--3
   .-p--3
-    button#popover-1(data-popover-content='<div class="-px--2 -py--1"><h5>Popover!</h5><p>This is a popover content triggered by a button. </p></div>').a-btn Popover
+    button#popover-1(data-popover-content='<div class="m-popover__content"><p class="m-popover__title">Popover!</p><p class="m-popover__text">This is a popover content.</p></div>').a-btn Popover
```
Comment (jllr) [inline src/website/views/javascript/popover.pug]: @janibolkvadze please review this page. It looks like markup is not updated

File: src/website/layouts/partials/stencilDoc.pug [hunk 3]
```diff
@@ -76,41 +76,40 @@ mixin events(events)
...
+                    if method.returns
+                      p.-text Type: #{method.returns.type}
+                  td
+                    div
+                      each parameter in method.parameters
+                        |#{parameter.name}:#{code=parameter.type}
+                        if parameter.docs
+                          div
+                            | Description:
+                            code=parameter.docs
```
Comment (jllr) [inline src/website/layouts/partials/stencilDoc.pug]: @janibolkvadze when there are not parameters it leaves a blank space. It is confusing, we should write something (ie: - )

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/431/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/432/ for more details.

**Pull request merged**